service/service.go

```diff
@@ -213,6 +202,7 @@ func buildLoggingMiddleware(logsous chan RequestEvent) func(http.Handler) http.H
 			userID := "-"
 			tokenString := r.Header.Get("X-Session-Token")
 			if tokenString != "" {
+				dataAccessLayer, _ := dataaccess.Get()
```

Comment:
If this error isn't handled, could the following line panic?

Reply:
This middle exists entirely for the purposes of logging who did what, and it's entirely possible for a non-nil error to be returned during normal functioning whenever an API request is made without a valid token. The behavior to log a `-` in place of the user name when the user is unknown is correct.

tl;dr -- the middleware shouldn't panic whenever an unknown user tries to interact with the API 🙂

Reply:
**BUT** on further reflection, I can make this more robust!

Reply:
Okay, ignore everything above because I was chasing the wrong point.

BUT -- I'd still rather not panic because that's hardly ever the helpful thing to do. Instead, the middleware logs an error and returns a status 500. This allows the operator to identify an error rate spike and also to correct a potentially broken config, without messy manics in handler processes.

How's that look? 🙂